vendor.go

```diff
@@ -160,6 +156,23 @@ func vendorPackages(depsGoHome string, projectdir string) {
 	})
 }
 
+func ignorePath(path string, isdir bool) (bool, error) {
+	base := filepath.Base(path)
+	if isdir {
+		if base == "vendor" || base == "testdata" {
+			return true, filepath.SkipDir
+		}
+	}
+	if base[0] == '.' || base[0] == '_' {
+		if isdir {
+			return true, filepath.SkipDir
+		}
+		return true, nil
+	}
+	ext := filepath.Ext(path)
+	return ext != ".go" && ext != ".s" && ext != ".c" && ext != ".h", nil
+}
```

Comment:
The function could be something like this, and then we would not need to check if it is a dir outside it (I think):

```go
func ignorePath(path string, isdir bool) (bool, error) {
	if isdir {
                base := filepath.Base(path)
		if base == "vendor" || base == "testdata" || base[0] == '.' || base[0] == '_' {
			return true, filepath.SkipDir
		}
                return true, nil
	}
	ext := filepath.Ext(path)
	return ext != ".go" && ext != ".s" && ext != ".c" && ext != ".h", nil
}
```

Reply:
We have to treat files different than directories because of this:
https://github.com/golang/go/blob/9852b4b093163c3d7bc28a2917f90eb297bbaa2e/src/path/filepath/path.go#L350

If it's a file and we return SkipDir, then walk() skips remaining files in the file's directory.

Reply:
But what this has to do with the code I proposed ? I just changed it to skip the same dirs as before + not skip the dir if it does not fit the pattern. The file logic is the same..no ?

Reply:
No, the logic is a little different. This way it doesn't skip files that starts with '.' or '_'.

Reply:
Ah yeah..true. We actually ignore + skip if it is a dir that starts with that and just ignore the file if it is a file. Sorry for the noise =P